packages/@aws-cdk/aws-stepfunctions-tasks/lib/sagemaker/create-processing-job.ts

```diff
@@ -194,18 +187,6 @@ export class SageMakerCreateProcessingJob extends sfn.TaskStateBase implements i
       maxRuntime: Duration.hours(1),
     };
 
-    props.processingInputs?.forEach(input => {
-      if (input.s3Input && input.datasetDefinition || (!input.s3Input && !input.datasetDefinition)) {
-        throw new Error('ProcessingInput must contain exactly one of either S3Input or DatasetDefinition');
-      }
-
-      const athenaDefinition = input.datasetDefinition?.athenaDatasetDefinition;
-      const redshiftDefinition = input.datasetDefinition?.athenaDatasetDefinition;
-      if (!input.s3Input && ((athenaDefinition && redshiftDefinition) || (!athenaDefinition && !redshiftDefinition))) {
-        throw new Error('DatasetDefinition must contain exactly one of either AthenaDatasetDefinition or RedshiftDatasetDefinition');
-      }
-    });
```

Comment:
One clear advantage of leaning further on the type system: This kind of validation is no longer necessary as we can't have a single processing input that defines properties for different datasets simultaneously (such as one input specifying properties for both an athena and redshift dataset definition).